tests/links/37e91a9f-b04e-5313-bbc0-4c394406247e_2021-01-01T00:00:00.000000Z_90bf705c-d891-5ce2-aa54-094488b445c6.json

```diff
@@ -0,0 +1,80 @@
+{
+    "describedBy": "https://schema.staging.data.humancellatlas.org/system/3.0.0/links",
+    "links": [
+        {
+            "inputs": [
+                {
+                    "input_id": "af3549f4-8632-58c2-9821-c6249c622b0a",
+                    "input_type": "cell_suspension"
+                }
+            ],
+            "link_type": "process_link",
+            "outputs": [
+                {
+                    "output_id": "f0c8ac7c-1c4a-5c9c-9286-c8b16208f9ae",
+                    "output_type": "analysis_file"
+                }
+            ],
+            "process_id": "d7b8cbff-aee9-5a05-a4a1-d8f4e720aee7",
+            "process_type": "process",
+            "protocols": [
+                {
+                    "protocol_id": "811adf54-7f27-5aec-9ed7-c6084c4c3a1b",
+                    "protocol_type": "analysis_protocol"
+                },
+                {
+                    "protocol_id": "767282b2-2fc1-5559-b020-29686b810206",
+                    "protocol_type": "library_preparation_protocol"
+                }
+            ]
+        },
+        {
+            "inputs": [
+                {
+                    "input_id": "3bee0c9a-aced-52f2-b97f-89518f826b36",
+                    "input_type": "specimen_from_organism"
+                }
+            ],
+            "link_type": "process_link",
+            "outputs": [
+                {
+                    "output_id": "af3549f4-8632-58c2-9821-c6249c622b0a",
+                    "output_type": "cell_suspension"
+                }
+            ],
+            "process_id": "a3843530-046c-5ec0-bbf4-a4f0929b60af",
+            "process_type": "process",
+            "protocols": [
+                {
+                    "protocol_id": "46435803-7ee8-5157-8569-8a61ad50c315",
+                    "protocol_type": "enrichment_protocol"
+                }
+            ]
+        },
+        {
+            "inputs": [
+                {
+                    "input_id": "5bc2bf2f-eb98-5510-a386-172cef38f2f2",
+                    "input_type": "donor_organism"
+                }
+            ],
+            "link_type": "process_link",
+            "outputs": [
+                {
+                    "output_id": "3bee0c9a-aced-52f2-b97f-89518f826b36",
+                    "output_type": "specimen_from_organism"
+                }
+            ],
+            "process_id": "37e91a9f-b04e-5313-bbc0-4c394406247e",
```

Comment:
@hannes-ucsc

Reply:
@hannes-ucsc the links.json reuses the assay process UUID for that subgraph by design.

Reply:
This is happening in production as well ever since DCP2 MVP release.

Reply:
The general assumption with entity IDs (especially UUID) is that they uniquely identify an entity. It would have been better to send that through a UUID v5 hash. So `links_id := UUIDv5(process_id)` instead of `links_id := process_id`.

> This has been happening in production as well ever since DCP2 MVP release.

I just confirmed with BQ that this is the case. It doesn't cause any issues with Azul, but that is only by coincidence, I could have easily written code that puts entities and links in a dictionary keyed by their ID.

Since what I objected to is actual production behavior, we can leave it as is. But we need to document this quirk:

https://github.com/HumanCellAtlas/dcp2/issues/20